rasa/core/channels/botframework.py

```diff
@@ -208,13 +208,23 @@ def blueprint(self, on_new_message):
                         postdata["recipient"],
                         postdata["serviceUrl"],
                     )
-
-                    user_msg = UserMessage(
-                        postdata["text"],
-                        out_channel,
-                        postdata["from"]["id"],
-                        input_channel=self.name(),
-                    )
+                    
+                    if postdata.get('attachments'):
+                        user_msg = UserMessage(
+                            text=(postdata['text'] if postdata.get('text') 
+                                  else ""),
+                            parse_data={"attachement": postdata['attachments'][0]},
```

Comment:
I don't think `parse_data` is the right place for an attachment. it should contain the result of an intent prediction

Reply:
You're right! Awesome that UserMessage now support metadata. It works like a charm!